river/preprocessing/one_hot.py

```diff
@@ -141,3 +231,60 @@ def transform_one(self, x, y=None):
                 oh[f"{i}_{xi}"] = 1
 
         return oh
+
+    # Mini-batch methods
+
+    @staticmethod
+    def _encode_1d(data, prefix, categories=None, sparse=False):
+        # REFLECTION: river often descends onto numpy level for computations
```

Comment:
Kinda referes to the `transform_many` situation where multiple `pd.DataFrame` objects are created and written into a list  and then concatenated. It's ok for `pandas` library to be like that but for `river` it would be better to descend to `numpy` level and at the `_encode_1d` level output instead a tuple of `np.ndarray` + column name ordered collection (e.g. list). Then at the `transform_many` concatenate the arrays and create `pd.DataFrame` only then. 

Haven't come up with a pretty way to do this or just didn't have time. Help appreciated.

Reply:
Nvm, got an idea, see commits below